internal/webhook/kubernetes/warehouse/webhook_test.go

```diff
@@ -261,8 +261,9 @@ func TestValidateSpec(t *testing.T) {
 							RepoURL: "bogus",
 						},
 						Image: &kargoapi.ImageSubscription{
-							SemverConstraint: "bogus",
-							Platform:         "bogus",
+							ImageSelectionStrategy: kargoapi.ImageSelectionStrategySemVer,
```

Comment:
I don't mind this line having been added for clarity, but just as a fyi, when no strategy is specified, behavior default to the semver strategy.

Reply:
> If I remove the strategy specified, it defaults to an empty string, due to which the test case will fail since validateImageSub() doesn't handle a scenario when a strategy has not been specified.

If you made changes to the validation method that are supposed to be non-breaking and it broke this test, then you should fix the method instead of altering the test case.

As it stands, your changes to the validation method are going to break everyone who is not explicitly specifying a selection strategy and the test was trying to tell you that.

Reply:
Please take a look now.